trunk/SUAVE/Components/Energy/Distributors/HTS_DC_Dynamo_Basic.py

```diff
@@ -0,0 +1,128 @@
+## @ingroup Components-Energy-Distributors
+# HTS_DC_Dynamo_Basic.py
+#
+# Created:  Feb 2020,   K. Hamilton
+# Modified: Jan 2022,   S. Claridge
+
+# ----------------------------------------------------------------------
+#  Imports
+# ----------------------------------------------------------------------
+
+# suave imports
+import SUAVE
+import numpy as np
+from SUAVE.Components.Energy.Energy_Component import Energy_Component
+import matplotlib.pyplot as plt
+
+# ----------------------------------------------------------------------
+#  HTS DC Dynamo Class
+# ----------------------------------------------------------------------
+
+## @ingroup Components-Energy-Distributors
+class HTS_DC_Dynamo_Basic(Energy_Component):
+    """ Basic HTS Dynamo model for constant current DC coil operation at constant cryogenic temperature.
+
+        Assumptions:
+        HTS Dynamo is operating at rated temperature and output current.
+
+        Source:
+        None
+    """
+
+    def __defaults__(self):
+        """ This sets the default values.
+    
+            Assumptions:
+            None
+    
+            Source:
+            N/A
+    
+            Inputs:
+            None
+    
+            Outputs:
+            None
+    
+            Properties Used:
+            None
+            """         
+        
+        self.efficiency             =   0.0      # [W/W]
+        self.mass_properties.mass   =   0.0      # [kg] 
+        self.rated_current          =   0.0      # [A]
+        self.rated_RPM              =   0.0      # [RPM]
+        self.rated_temp             =   0.0      # [K]
+    
+    def shaft_power(self, cryo_temp, hts_current, power_out):
+        """ The shaft power that must be supplied to the DC Dynamo supply to power the HTS coils.
+            Assumptions:
+                HTS Dynamo is operating at rated temperature.
+                
+            Source:
+                N/A
+
+            Inputs:
+                cryo_temp           [K]
+                current             [A]
+                power_out           [W]
+
+            Outputs:
+                power_in            [W]
+                cryo_load           [W]
+
+            Properties Used:
+                None
+        """
+
+
+        #Adjust efficiency according to the rotor current 
+        current    = np.array(hts_current)
+        efficiency = self.efficiency_curve(current)
+
+        # Create output arrays. The hts dynamo input power is assumed zero if the output power is zero, this may not be true for some dynamo configurations however the power required for zero output power will be very low.
+        # Similarly, the cryo load will be zero if no dynamo effect is occuring.
+        power_in   = np.zeros_like(power_out)
+        cryo_load  = np.zeros_like(power_out)
+
+        power_in   = np.array(power_out/efficiency)
```

Comment:
the zeroed out arrays are getting overwritten here. If you want to reset power_in to be zero if power_out is zero it would look like this:

power_in = np.array(power_out/efficiency)
power_in[power_out==0.] = 0